src/create_disk.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+
+usage() {
+	echo "create_disk -d disk -o ostree"
+}
+
+export PATH=$PATH:/sbin:/usr/sbin
+
+getpart() {
+	# getpart /dev/loop0 1 -> /dev/loop0p1
+	# getpart /dev/sda 1 -> /dev/sda1
+	last="${1: -1}"
+	if [ $last -q $last 2>/dev/null ]; then
+		echo "${1}p${2}"
+	else
+		echo "${1}${2}"
+	fi
+}
+
+rc=0
+TEMP=$(getopt -o "d:o:r:" --long "disk:,ostree:,ref:" -- "$@") || rc=$?
+if [ "$rc" -ne 0 ]; then
+	usage
+	exit 1
+fi
+
+eval set -- "$TEMP"
+
+while :
+do
+	case "$1" in
+		"-d"|"--disk")
+			shift
+			disk="$1"
+			shift
+			;;
+		"-o"|"--ostree")
+			shift
+			ostree="$1"
+			shift
+			;;
+		"-r"|"--ref")
+			shift
+			ref="$1"
+			shift
+			;;
+		--)
+			shift
+			break
+			;;
+		*)
+			echo "Error parsing args"
+			usage
+			exit 1
+			;;
+	esac
+done
+
+[ -z "$disk" ] || [ -z "$ostree" ] && {
+	usage
+	exit 1
+}
+
+set -e
+
+script_dir=$(dirname $(readlink -f "$0"))
+# partition and create fs
+sgdisk -Z $disk \
+	-n 1:0:+128M -c 1:boot \
+	-n 2:0:+128M -c 2:EFI-SYSTEM -t 1:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
+	-n 3:0:+128M -c 3:BIOS-BOOT  -t 2:21686148-6449-6E6F-744E-656564454649 \
+	-n 4:0:0     -c 4:root       -t 3:4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
+sgdisk -p $disk
+
+# HACK ALERT - wait for partition rescans
+sleep 2
+# FIXME ostree needs symlinks
+mkfs.ext2 "$(getpart ${disk} 1)" -L boot
+mkfs.fat "$(getpart ${disk} 2)" -n EFI-SYSTEM
+mkfs.xfs "$(getpart ${disk} 4)"  -L root
```

Comment:
Currently we also force on `-m reflink=1` but eh...I guess we can just take that default from mkfs in the near future.